python/cudf/dataframe/categorical.py

```diff
@@ -173,13 +173,18 @@ def to_pandas(self, index=None):
     def to_arrow(self):
         mask = None
         if self.has_null_mask:
-            # Necessary because PyArrow doesn't support from_buffers for
-            # DictionaryArray yet
-            mask = cudautils.expand_mask_bits_logical_not(
-                len(self),
-                self.nullmask.mem
-            ).copy_to_host().astype('bool')
-        indices = pa.array(self.cat().codes.data.mem.copy_to_host(), mask=mask)
+            mask = pa.py_buffer(self.nullmask.mem.copy_to_host())
+        indices = self.cat().codes.to_arrow()
```

Comment:
If there's nulls wouldn't this already have the nullmask as part of the arrow array that's created? This would effectively copy the nullmask twice.

Reply:
If this doesn't have the nullmask already then we should just do `indices_data = pa.py_buffer(self.cat().codes.data.mem.copy_to_host())`

Reply:
Good point, it looks like the nullmask is already being taken into account by the CategoricalAccessor.codes getter:
https://github.com/rapidsai/cudf/blob/db21c281d45ac714f853d86513277a4b4e89b325/python/cudf/dataframe/categorical.py#L32-L40